plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/command/edit/EditCommandsPresenter.java

```diff
@@ -68,16 +68,14 @@
     private final MachineLocalizationConstant          machineLocale;
     private final CoreLocalizationConstant             coreLocale;
     private final Provider<SelectCommandComboBoxReady> selectCommandActionProvider;
-
     private final Set<ConfigurationChangedListener> configurationChangedListeners;
-
+    /** HashSet of the existing command names. */
+    private final HashSet<String>                                commandNames;
```

Comment:
Why not declare it as *Set<String> commandNames*? You should programm to an interface